### PR TITLE
Add opt out get and create support

### PR DESCRIPTION
**Pull request by Codesleuth**
Adds the ability to get and create opt out resources.

Includes some changes and improvements to the existing tests to increase coverage for regression protection.

Usage:

* Get single opt out

    ```java
    UserPassword userPassword = new UserPassword("YourUsername", "YourPassword");
    OptOutService optOutService = new OptOutServiceImpl(new BasicAuthenticator(userPassword));
    OptOutResponse optout = optOutService.getOptOut("b6a39581-9cfc-40ea-9b40-9320444cf49d");
    ```

* Get paged opt outs

    ```java
    UserPassword userPassword = new UserPassword("YourUsername", "YourPassword");
    OptOutService optOutService = new OptOutServiceImpl(new BasicAuthenticator(userPassword));
    OptOutCollectionResponse optouts = optOutService.getOptOuts(1, 15);
    ```

* Create an opt out

    ```java
    UserPassword userPassword = new UserPassword("YourUsername", "YourPassword");
    OptOutService optOutService = new OptOutServiceImpl(new BasicAuthenticator(userPassword));

    FromAddress fromAddress = new FromAddress();
    fromAddress.setPhoneNumber("99887744556322");

    OptOutRequest request = new OptOutRequestImpl();
    request.setAccountReference("EX006789");
    request.setFromAddress(fromAddress);

    OptOutResponse optout = optOutService.createOptOut(request);
    ```

### CODE REVIEW

File: esendex-java-sdk.iml [hunk 1]
```diff
@@ -12,16 +12,16 @@
...
     <orderEntry type="library" name="Maven: xmlpull:xmlpull:1.1.3.1" level="project" />
     <orderEntry type="library" name="Maven: xpp3:xpp3_min:1.1.4c" level="project" />
-    <orderEntry type="library" name="Maven: junit:junit:4.9" level="project" />
-    <orderEntry type="library" name="Maven: org.hamcrest:hamcrest-core:1.1" level="project" />
-    <orderEntry type="library" name="Maven: com.pyruby:java-stub-server:0.12" level="project" />
-    <orderEntry type="library" name="Maven: org.mortbay.jetty:jetty:6.1.26" level="project" />
-    <orderEntry type="library" name="Maven: org.mortbay.jetty:jetty-util:6.1.26" level="project" />
-    <orderEntry type="library" name="Maven: org.mortbay.jetty:servlet-api:2.5-20081211" level="project" />
-    <orderEntry type="library" name="Maven: javax.servlet:servlet-api:2.5" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: junit:junit:4.12" level="project" />
```
Comment (jackthorley) [inline esendex-java-sdk.iml]: :+1:

File: README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Esendex Java SDK
+================
+
+[![Build Status](https://travis-ci.org/esendex/esendex-java-sdk.svg?branch=master)](https://travis-ci.org/esendex/esendex-java-sdk)
```
Comment (moretonb) [inline README.md]: :yellow_heart:

File: src/main/java/esendex/sdk/java/model/domain/impl/OptOutResponseAssembler.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        return response;
+    }
+
+    public OptOutCollectionResponse createCollectionResponse(OptOutCollectionResponseDto col) {
+
+        List<OptOutResponse> optouts = new ArrayList<>();
+        if (col.getOptouts() != null) {
+            for (OptOutResponseDto dto : col.getOptouts()) {
+                OptOutResponseAssembler assembler = new OptOutResponseAssembler();
```
Comment (moretonb) [inline src/main/java/esendex/sdk/java/model/domain/impl/OptOutResponseAssembler.java]: err, isn't this a little unnecessary?

File: src/main/java/esendex/sdk/java/model/domain/impl/OptOutResponseAssembler.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                optouts.add(assembler.createResponse(dto));
+            }
+        }
+
+        OptOutCollectionResponseImpl response = new OptOutCollectionResponseImpl(optouts);
+        response.setCount(col.getCount());
+        response.setStartIndex(col.getStartindex());
+        response.setTotalCount(col.getTotalcount());
+        response.setOptouts(optouts);
```
Comment (moretonb) [inline src/main/java/esendex/sdk/java/model/domain/impl/OptOutResponseAssembler.java]: isn't this set by the constructor?

File: src/main/java/esendex/sdk/java/service/resource/optout/RetrieveOptOutResource.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        extends XmlResponderResource<OptOutResponseDto> {
+
+    /**
+     * Instantiates a new retrieve contact resource.
+     *
+     * @param auth the authenticator
+     * @param id   the opt out id
+     */
+    public RetrieveOptOutResource(Authenticator auth, String id) {
+        super(auth, null, id, null, "1.0");
```
Comment (moretonb) [inline src/main/java/esendex/sdk/java/service/resource/optout/RetrieveOptOutResource.java]: version doesn't need to be explicit, can use the other constructor

File: src/test/java/com/esendex/java/sdk/OptOutService/CreateOptOutTest.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Test
+    public void thenTheOptOutIsCreated() {
+        server.verify();
+        assertEquals(expectedRequestBody, stubMethod.bodyString());
+    }
+
+    @Test
+    public void thenTheCreatedOptOutIsReturned() {
+        assertEquals("5d3c5c2d-ae35-4001-bf64-8893c091cb7f", result.getId());
```
Comment (moretonb) [inline src/test/java/com/esendex/java/sdk/OptOutService/CreateOptOutTest.java]: these raw strings could be in fields?

File: src/test/java/com/esendex/java/sdk/SentService/GetMessageTest.java [hunk 7]
```diff
@@ -11,62 +11,84 @@
...
+    }
+
+    @Test
+    public void thenTheMessageHeaderIsRetrieved() {
+        server.verify();
     }
 
     @Test
     public void thenTheMessagePropertiesAreSet() {
+        assertEquals("CDEB3533-1F76-46D7-A2A9-0DAF8290F7FC", result.getId());
```
Comment (moretonb) [inline src/test/java/com/esendex/java/sdk/SentService/GetMessageTest.java]: :+1:

File: src/test/java/com/esendex/java/sdk/integration/inboxservice/InboxServiceDeleteIT.java [hunk 8]
```diff
@@ -1,37 +0,0 @@
...
-        {
-            messageIdToDelete = messages.getMessages().get(0).getId();
-            assertTrue(inboxService.deleteMessage(messageIdToDelete));
-        }
-
-        messagesAfterDelete = inboxService.getMessages();
-    }
-
-    @Test
-    public void thenTheMessageIsNotInTheMessagesRetrievedAfterTheDelete() {
```
Comment (moretonb) [inline src/test/java/com/esendex/java/sdk/integration/inboxservice/InboxServiceDeleteIT.java]: lul

**Comment by moretonb** [comment]
I can't see a new addition to the IServiceFactory

**Pull request closed without merging**